ui-v2/src/components/flow-runs/flow-runs-list/flow-runs-filters/work-pool-filter.tsx

```diff
@@ -0,0 +1,151 @@
+import { useQuery } from "@tanstack/react-query";
+import { useDeferredValue, useMemo, useState } from "react";
+import {
+	buildFilterWorkPoolsQuery,
+	type WorkPool,
+} from "@/api/work-pools/work-pools";
+import { Checkbox } from "@/components/ui/checkbox";
+import {
+	Combobox,
+	ComboboxCommandEmtpy,
+	ComboboxCommandGroup,
+	ComboboxCommandInput,
+	ComboboxCommandItem,
+	ComboboxCommandList,
+	ComboboxContent,
+	ComboboxTrigger,
+} from "@/components/ui/combobox";
+import { Typography } from "@/components/ui/typography";
+
+const MAX_WORK_POOLS_DISPLAYED = 2;
+
+type WorkPoolFilterProps = {
+	selectedWorkPools: Set<string>;
+	onSelectWorkPools: (workPools: Set<string>) => void;
+};
+
+export const WorkPoolFilter = ({
+	selectedWorkPools,
+	onSelectWorkPools,
+}: WorkPoolFilterProps) => {
+	const [search, setSearch] = useState("");
+	const deferredSearch = useDeferredValue(search);
+
+	const { data: workPools = [] } = useQuery(
+		buildFilterWorkPoolsQuery({
+			limit: 100,
+			offset: 0,
+		}),
+	);
+
+	const { data: selectedWorkPoolsData = [] } = useQuery(
+		buildFilterWorkPoolsQuery(
+			{
+				work_pools:
+					selectedWorkPools.size > 0
+						? {
+								operator: "and_",
+								name: { any_: Array.from(selectedWorkPools) },
+							}
+						: undefined,
+				limit: selectedWorkPools.size || 1,
+				offset: 0,
+			},
+			{ enabled: selectedWorkPools.size > 0 },
+		),
+	);
+
+	const handleSelectWorkPool = (workPoolName: string) => {
+		const updatedWorkPools = new Set(selectedWorkPools);
+		if (selectedWorkPools.has(workPoolName)) {
+			updatedWorkPools.delete(workPoolName);
+		} else {
+			updatedWorkPools.add(workPoolName);
+		}
+		onSelectWorkPools(updatedWorkPools);
+	};
+
+	const handleClearAll = () => {
+		onSelectWorkPools(new Set());
+	};
+
+	const renderSelectedWorkPools = () => {
+		if (selectedWorkPools.size === 0) {
+			return "All work pools";
+		}
+
+		const selectedWorkPoolNames = selectedWorkPoolsData
+			.filter((workPool) => selectedWorkPools.has(workPool.name))
+			.map((workPool) => workPool.name);
```

Comment:
This is unecessary because the component will be passed a list of work pool names already in `selectedWorkPools`.